fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java

```diff
@@ -420,6 +420,12 @@ private void finalizeCommand() throws IOException {
 
         MysqlChannel channel = ctx.getMysqlChannel();
         channel.sendAndFlush(packet);
+        // note(wb) we should write profile after return result to mysql client
+        // because write profile maybe take too much time
+        // explain query stmt do not have profile
+        if (executor != null && !executor.getParsedStmt().isExplain()) {
+            executor.writeProfile(true);
```

Comment:
Whether instanceof queryStmt judged before, do you want to expand the profile scope?

Reply:
I want to write profile after query result return to client.

Reply:
> I want to write profile after query result return to client.
I know the idea, but I mean the side effect of this move is to expand the scope of the profile. Such as insert, use etc. also write profile.

Reply:
> I want to write profile after query result return to client.

I know the idea, but I mean the side effect of this move is to expand the scope of the profile. Such as insert, use etc. also write profile.

Reply:
It seems that it didn't expand the scope of the profile

Reply:
Do you mean this may cause profile inaccurate?